firedrake/slate/slate.py

```diff
@@ -140,6 +141,42 @@ def inv(self):
     def T(self):
         return Transpose(self)
 
+    def block(self, test_idx, trial_idx=None):
```

Comment:
I chose `block` because that seemed to be the logical choice for name here. I want "a block of a matrix over argument indices (...)" I'm not opposed to renaming it, but there was reason and thought into this.

I think `split` could work. So this probably means this method should return several `Block` objects corresponding to the function arguments provided?